protostar/argument_parser/argument_parser_facade.py

```diff
@@ -27,20 +27,32 @@
 
 
 def parse_collection_arg(arg: list[Union[int, dict[str, Any]]]):
-    arg_type: Optional[type] = None
+    if not arg:
+        return arg
+
+    dict_present = False
     result = arg
     for input_item in result:
-        if arg_type is None:
-            arg_type = type(input_item)
-            continue
-        if not isinstance(input_item, arg_type):
-            raise InconsistentInputTypesException()
+        if isinstance(input_item, dict):
+            dict_present = True
+            break
+    if not isinstance(arg[0], dict) and dict_present:
```

Comment:
So if we have a named arg present (represented by a `dict`), the args in general have to start with a named arg. Then, we can have just regular args (that would be interpreted as a list) but they cannot start with a list.